docs/modules/ROOT/pages/how-to-guides/Import-code/proc_importing_code.adoc

```diff
@@ -0,0 +1,207 @@
++
+**Solutions** 
++
+. Make sure that your `devfile.yaml` points to the correct directories for both your Dockerfile and your Kubernetes YAML file. 
+. Make sure that neither of these resources is in a private repository that requires access authentication.
+. Make sure that your devfile is in one of these expected locations:
++
+[#expected devfile locations] 
+.Expected devfile locations
+** devfile.yaml
+** .devfile.yaml
+** .devfile or devfile.yaml
+** .devfile or .devfile.yaml
+
+* **Your repository does not fit within the predefined {ProductName} supported runtime list.**
++
+Some repositories require custom build or deployment instructions; for example, a deployment that includes a Kubernetes resource other than a YAML file.
++
+**Solutions**
++
+[#custom build and deployment instructions]
+** **Configure a custom build.** Include a Dockerfile that can build your application, then save it to one of the <<expected Dockerfile locations>>. **Note:** If your build is custom but your deployment is not, you do not have to include a `devfile.yaml` file in your repository.
+** **Configure a custom deployment.** Include the following files in your repository to provide {ProductName} with custom deployment instructions:
+*** A standard Kubernetes YAML file
+*** A custom-build Dockerfile
+*** A devfile that points to your Kubernetes YAML file and Dockerfile. **Note:** Make sure that your devfile is in one of the <<expected devfile locations>>.
+
+* **{ProductName} couldn't detect a port number.**
++
+{ProductName} detects your application port number when it's looking for your repository components. 
++
+**Solution:** From the **Review and configure for deployment** view, in the **Build & deploy configuration** section, make sure your target port number is correct.
+
+* **{ProductName} couldn't detect the appropriate runtime.**
++
+**Solution:** Select a predefined runtime type from the **Review and configure for deployment** view. Make sure you meet all of the {ProductName} <<repository requirements>> for the runtime you select, or configure your own <<custom build and deployment instructions>>.
+
+== Reviewing and configuring components
+
+[role="_abstract"]
+If you imported your own code, from the **Review and configure for deployment** view, make sure to define, review, or change your application and component settings before you finish importing. You can rename a component here, too. 
+
+. Make sure that **Component name** is correct. Click the GitHub link under this field if you want to check your repository URL.
+. Confirm that {ProductName} detected the right **Runtime**. If so, you should notice that the suggested deployment options are appropriate. If you want to change your runtime, expand the menu next to your runtime name and make a selection.
+
+[IMPORTANT]
+====
+If you change the runtime, your component builds and deploys using the runtime devfile instead of the component devfile or Dockerfile.
```

Comment:
I think the text that we came up with during the last review was "If you change the runtime, your component builds and deploys using the runtime devfile instead of the repository devfile or Dockerfile."